control-plane/connect-inject/endpoints_controller_test.go

```diff
@@ -344,6 +371,43 @@ func TestProcessUpstreams(t *testing.T) {
 				},
 			},
 			consulNamespacesEnabled: false,
+			consulPartitionsEnabled: false,
+		},
+		{
+			name: "multiple upstreams with consul namespaces, partitions and datacenters",
+			pod: func() *corev1.Pod {
+				pod1 := createPod("pod1", "1.2.3.4", true, true)
+				pod1.Annotations[annotationUpstreams] = "upstream1:1234, upstream2.bar:2234, upstream3.foo.baz:3234:dc2"
+				return pod1
+			},
+			configEntry: func() api.ConfigEntry {
+				ce, _ := api.MakeConfigEntry(api.ProxyDefaults, "pd")
+				pd := ce.(*api.ProxyConfigEntry)
+				pd.MeshGateway.Mode = "remote"
+				return pd
+			},
+			expected: []api.Upstream{
+				{
+					DestinationType: api.UpstreamDestTypeService,
+					DestinationName: "upstream1",
+					LocalBindPort:   1234,
+				},
+				{
+					DestinationType:      api.UpstreamDestTypeService,
+					DestinationName:      "upstream2",
+					DestinationNamespace: "bar",
+					LocalBindPort:        2234,
+				}, {
+					DestinationType:      api.UpstreamDestTypeService,
+					DestinationName:      "upstream3",
+					DestinationNamespace: "foo",
+					DestinationPartition: "baz",
+					LocalBindPort:        3234,
+					Datacenter:           "dc2",
+				},
+			},
+			consulNamespacesEnabled: true,
+			consulPartitionsEnabled: true,
```

Comment:
Q: Is partitions with namespaces disabled supported?

Reply:
🤔 it should be. ill write a test for it. Everything should land in the default namespace in that case.
Should we error if the namespace provided is non-default?

Reply:
No I think it's fine as is. Just might be weird for folks to have to use `svc.ns.part` if they don't have namespaces enabled but I think that's fine.